woodchipper/processors.py

```diff
@@ -31,3 +31,43 @@ def __call__(self, logger: logging.Logger, method: str, event_dict: dict):
         except (OSError, json.JSONDecodeError):
             pass
         return event_dict
+
+
+class DatadogTraceProcessor:
+    """
+    Adds Datadog trace information via the ddtrace library if it is installed
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def __call__(self, logger: logging.Logger, method: str, event_dict: dict) -> dict:
+        # Adapted from
+        # https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/python/#no-standard-library-logging
+        try:
+            import ddtrace
+            from ddtrace import tracer
+        except ImportError:
+            # ddtrace is an optional dependency
+            # if not installed, no-op the event
+            return event_dict
```

Comment:
Why be defensive here? If the user did `from woodchipper.processors import DataDogTraceProcessor` doesn't that indicate a lot of intent to interop with DD?

Seems worse to fail silently imo.

Reply:
If an app using the built-in logging configs (`JSONLogToStdout`) then it _isn't_ explicitly importing the processor since it's included by default. If it's a more explicit opt-in then yeah it makes a bit more sense to be less defensive. For what it's worth I was roughly following how `SentryJsonProcessor` behaves; if `structlog_sentry` isn't installed then it will be a no-op processor.

Also I made ddtrace an optional dependency since it brings in quite a few other dependencies with it. For any apps that aren't instrumented with Datadog (yet or otherwise) that's quite a few unnecessary libraries to bring in that don't do anything.

Reply:
Feels like `JSONLogToStdout` is becoming a godclass :/ . fwiw I think the same thing about the Sentry processor being auto included too.

Good point with SentryJsonProcessor behavior being the same.

Reply:
Hmm not sure I completely agree. `JSONLogToStdout` is a fantastic low-code configuration with sensible defaults for most apps. It's really nice that it Just Works and let's me need think less about making sure my logging system is in-line with all of the other apps. Just my opinion tho

Reply:
Yeah I think that's a good example of jag haven't good intuition for ergonomics of use that it can be added like that.

But if my application doesn't use DD or Sentry (we don't/won't, so it's a good example) or Sentry, I now have DD and Sentry looking stuff kinda in my codebase.

Vs an extension approach where you'd use a base processor that imported only from stdlib and structlib and allowed the library user to add the third parties they use.

Reply:
Counterargument: you _don't_ have Sentry or DD looking stuff in your codebase because it's all contained in the library. The library just goes "oh you aren't using DD/Sentry, w/e" or "oh I see DD/Sentry I'll do the thing you probably want to do." Guess it's kind of a convention-over-configuration approach?

This is definitely out of scope for this PR but you could have a factory function for generating a base configuration based on parameters you give it, like extra processors for Sentry/DD but it keeps all of the boilerplate-y stuff.

Reply:
@Ian288 Good point. Moved it to the top and just set a flag. Tested it locally and that works fine.